service/controller/azureconfig/handler/capzcrs/create.go

```diff
@@ -175,6 +176,16 @@ func (r *Resource) mapAzureConfigToAzureCluster(ctx context.Context, cr provider
 			},
 			ResourceGroup: key.ClusterID(&cr),
 			NetworkSpec: capzv1alpha3.NetworkSpec{
+				APIServerLB: capzv1alpha3.LoadBalancerSpec{
+					Name: fmt.Sprintf("%s-%s-%s", cr.Name, "API", "PublicLoadBalancer"),
```

Comment:
Not sure if they are used anywhere else, but what do you think about creating functions in the `key` package that return this string (and the `FrontendIPs` name too)?

Reply:
These are GS-specific requirements currently, would it be possible to have these `APIServerLB` fields in admission controller create defaulting? And then also have those values validated in admission controller, and block change/update of these values?
This would be in addition to what's done here, as the mapping is needed for existing CRs, but the mapping will maybe be gone soon-ish.

WDYT?